api/apps/v1alpha1/nemo_customizer_types.go

```diff
@@ -0,0 +1,925 @@
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	// NemoCustomizerConditionReady indicates that the NEMO CustomizerService is ready.
+	NemoCustomizerConditionReady = "Ready"
+	// NemoCustomizerConditionFailed indicates that the NEMO CustomizerService has failed.
+	NemoCustomizerConditionFailed = "Failed"
+
+	// NemoCustomizerStatusPending indicates that NEMO CustomizerService is in pending state
+	NemoCustomizerStatusPending = "Pending"
+	// NemoCustomizerStatusNotReady indicates that NEMO CustomizerService is not ready
+	NemoCustomizerStatusNotReady = "NotReady"
+	// NemoCustomizerStatusReady indicates that NEMO CustomizerService is ready
+	NemoCustomizerStatusReady = "Ready"
+	// NemoCustomizerStatusFailed indicates that NEMO CustomizerService has failed
+	NemoCustomizerStatusFailed = "Failed"
+
+	// SchedulerTypeVolcano indicates if the scheduler is volcano
+	SchedulerTypeVolcano = "volcano"
+	// SchedulerTypeRunAI indicates if the scheduler is run.ai
+	SchedulerTypeRunAI = "runai"
+)
+
+// NemoCustomizerSpec defines the desired state of NemoCustomizer
+type NemoCustomizerSpec struct {
+	Image          Image                        `json:"image,omitempty"`
+	Command        []string                     `json:"command,omitempty"`
+	Args           []string                     `json:"args,omitempty"`
+	Env            []corev1.EnvVar              `json:"env,omitempty"`
+	Labels         map[string]string            `json:"labels,omitempty"`
+	Annotations    map[string]string            `json:"annotations,omitempty"`
+	NodeSelector   map[string]string            `json:"nodeSelector,omitempty"`
+	Tolerations    []corev1.Toleration          `json:"tolerations,omitempty"`
+	PodAffinity    *corev1.PodAffinity          `json:"podAffinity,omitempty"`
+	Resources      *corev1.ResourceRequirements `json:"resources,omitempty"`
+	Expose         Expose                       `json:"expose,omitempty"`
+	LivenessProbe  Probe                        `json:"livenessProbe,omitempty"`
+	ReadinessProbe Probe                        `json:"readinessProbe,omitempty"`
+	StartupProbe   Probe                        `json:"startupProbe,omitempty"`
+	Scale          Autoscaling                  `json:"scale,omitempty"`
+	Metrics        Metrics                      `json:"metrics,omitempty"`
+
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default:=1
+	Replicas     int    `json:"replicas,omitempty"`
+	UserID       *int64 `json:"userID,omitempty"`
+	GroupID      *int64 `json:"groupID,omitempty"`
+	RuntimeClass string `json:"runtimeClass,omitempty"`
+
+	// CustomizerConfig stores the customizer configuration for training and models
+	// +kubebuilder:validation:MinLength=1
+	CustomizerConfig string `json:"customizerConfig"`
+
+	// Scheduler Configuration
+	Scheduler Scheduler `json:"scheduler,omitempty"`
+
+	// OpenTelemetry Settings
+	OpenTelemetry OTelSpec `json:"otel"`
```

Comment:
nit: this should be optional
```suggestion
	// OpenTelemetry Settings
	// +kubebuilder:validation:Optional
	OpenTelemetry *OTelSpec `json:"otel"`
```

Reply:
no, customizer doesn't function without OTel configuration